packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/library/StudioLibraryPublicationHandler.java

```diff
@@ -0,0 +1,210 @@
+import org.eclipse.sirius.web.application.library.services.api.ILibraryPublicationHandler;
+import org.eclipse.sirius.web.application.studio.services.library.api.IStudioLibraryDependencyCollector;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.Library;
+import org.eclipse.sirius.web.domain.boundedcontexts.library.services.api.ILibraryCreationService;
+import org.eclipse.sirius.web.domain.boundedcontexts.projectsemanticdata.services.api.IProjectSemanticDataSearchService;
+import org.eclipse.sirius.web.domain.boundedcontexts.semanticdata.SemanticData;
+import org.eclipse.sirius.web.domain.boundedcontexts.semanticdata.services.api.ISemanticDataCreationService;
+import org.eclipse.sirius.web.domain.services.IResult;
+import org.eclipse.sirius.web.domain.services.Success;
+import org.eclipse.sirius.web.domain.services.api.IMessageService;
+import org.springframework.data.jdbc.core.mapping.AggregateReference;
+import org.springframework.stereotype.Service;
+
+/**
+ * Handles the publication of libraries from studios.
+ *
+ * @author gdaniel
+ */
+@Service
+public class StudioLibraryPublicationHandler implements ILibraryPublicationHandler {
+
+    private final IEditingContextSearchService editingContextSearchService;
+
+    private final ISemanticDataCreationService semanticDataCreationService;
+
+    private final IResourceToDocumentService resourceToDocumentService;
+
+    private final IStudioLibraryDependencyCollector studioLibraryDependencyCollector;
+
+    private final ILibraryCreationService libraryCreationService;
+
+    private final IProjectSemanticDataSearchService projectSemanticDataSearchService;
+
+    private final IMessageService messageService;
+
+    public StudioLibraryPublicationHandler(IEditingContextSearchService editingContextSearchService, ISemanticDataCreationService semanticDataCreationService, IResourceToDocumentService resourceToDocumentService, IStudioLibraryDependencyCollector studioLibraryDependencyCollector, ILibraryCreationService libraryCreationService, IProjectSemanticDataSearchService projectSemanticDataSearchService, IMessageService messageService) {
+        this.editingContextSearchService = Objects.requireNonNull(editingContextSearchService);
+        this.semanticDataCreationService = Objects.requireNonNull(semanticDataCreationService);
+        this.resourceToDocumentService = Objects.requireNonNull(resourceToDocumentService);
+        this.studioLibraryDependencyCollector = Objects.requireNonNull(studioLibraryDependencyCollector);
+        this.libraryCreationService = Objects.requireNonNull(libraryCreationService);
+        this.projectSemanticDataSearchService = Objects.requireNonNull(projectSemanticDataSearchService);
+        this.messageService = Objects.requireNonNull(messageService);
+    }
+
+    @Override
+    public boolean canHandle(PublishLibrariesInput input) {
+        return Objects.equals(input.publicationKind(), "studio-all");
```

Comment:
ooh I see what this kind is about, I may rename this attribute but something like that was definitely necessary.